### PR TITLE
Package rules in proto response from current endpoint

### PR DESCRIPTION
**Pull request by robfletcher**

### CODE REVIEW

File: keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt [hunk 1]
```diff
@@ -124,18 +130,67 @@ class AmazonSecurityGroupHandler(
...
-            it.region = region
-            it.vpcName = vpcId?.let { cloudDriverCache.networkBy(it).name }
-            it.description = description
+          .also { builder ->
+            builder.name = name
+            builder.accountName = accountName
+            builder.region = region
+            builder.vpcName = vpcId?.let { cloudDriverCache.networkBy(it).name }
+            builder.description = description
+            inboundRules.forEach { rule ->
```
Comment (robfletcher) [inline keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt]: This whole bit with the inbound rules was missing so the current and desired states never match.

File: keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/model/SecurityGroup.kt [hunk 2]
```diff
@@ -41,7 +41,8 @@ data class SecurityGroup(
   data class SecurityGroupRuleReference(
     val name: String,
     val accountName: String?,
-    val region: String?
+    val region: String?,
+    val vpcId: String?
```
Comment (robfletcher) [inline keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/model/SecurityGroup.kt]: Need to resolve the VPC to map to the proto response

File: keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt [hunk 3]
```diff
@@ -124,18 +130,67 @@ class AmazonSecurityGroupHandler(
...
+              builder.inboundRuleList.add(
+                SecurityGroupRule.newBuilder().also { builder ->
+                  val referencedGroup = rule.securityGroup
+                  val cidrRange = rule.range
+                  when {
+                    referencedGroup != null && (referencedGroup.vpcId == vpcId && referencedGroup.name == name) -> {
```
Comment (robfletcher) [inline keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt]: I believe this is the correct condition for a self-referencing rule

File: keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt [hunk 4]
```diff
@@ -124,18 +130,67 @@ class AmazonSecurityGroupHandler(
...
+                      builder.selfReferencingRuleBuilder.also { ruleBuilder ->
+                        ruleBuilder.protocol = rule.protocol
+                        rule.portRanges?.forEach {
+                          ruleBuilder.portRangeList.add(it.toProto())
+                        }
+                      }
+                    }
+                    referencedGroup != null && (referencedGroup.accountName != accountName || referencedGroup.region != region) -> {
```
Comment (robfletcher) [inline keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt]: If account or region differ from the parent security group, it's a cross-region rule.

File: keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt (hunk 1 above)
Comment (robzienert) [reply keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt]: 🤦‍♂️ oops

File: keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt (hunk 3 above)
Comment (robzienert) [reply keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt]: This looks sane, yes.

Comment (robfletcher) [reply keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt]: Specifically no need to check account / region as the vpcId would be different if those were.

**Pull request merged**